apps/wing/src/commands/test/snapshots.ts

```diff
@@ -0,0 +1,178 @@
+      return SnapshotMode.ASSERT;
+    } else {
+      return SnapshotMode.DEPLOY;
+    }
+  }
+
+  return options.snapshots;
+}
+
+export async function captureSnapshot(entrypoint: string, target: string, options: TestOptions) {
+  const snapshotMode = determineSnapshotMode(target, options);
+
+  // skip if snapshots are disabled
+  if (snapshotMode === SnapshotMode.NEVER) {
+    return;
+  }
+
+  const snapshotFile = `${entrypoint}.${target}.snap.md`;
+
+  await withSpinner(`Snapshot ${renderTestName(snapshotFile)}...`, async () => {
+    // we need to compile again in order for snapshots because we can't afford the snapshot to be
+    // based on a random root id (which is the default for tests)
+    const snapshotDir = await compile(entrypoint, options);
+
+    // take a snapshot of the synthesis output
+    const snapshot = await createMarkdownSnapshot(basename(snapshotFile), snapshotDir);
+
+    // update the snapshots if needed
+    switch (snapshotMode) {
+      case SnapshotMode.DEPLOY:
+      case SnapshotMode.UPDATE:
+        writeFileSync(snapshotFile, snapshot);
+        break;
+
+      case SnapshotMode.ASSERT:
+        if (!existsSync(snapshotFile)) {
+          throw new Error(
+            [`Snapshot file does not exist: ${snapshotFile}`, "", SNAPSHOTS_ERROR_HELP].join("\n")
+          );
+        }
+
+        const expectedSnapshot = readFileSync(snapshotFile, "utf-8");
+        if (expectedSnapshot !== snapshot) {
+          const actualFile = `${snapshotFile}.actual`;
+          writeFileSync(actualFile, snapshot);
+
+          const diff =
+            tryRenderDiff(snapshotFile, actualFile) ??
+            [` - Expected: ${snapshotFile}`, ` - Actual: ${actualFile}`].join("\n");
+
+          throw new Error(["Snapshot mismatch:", "", diff, "", SNAPSHOTS_ERROR_HELP].join("\n"));
+        }
+        break;
+    }
+  });
+}
+
+/**
+ * Uses the "diff" command to render a diff between two files. Returns `undefined` if the diff command
+ * is not available or if there was an error.
+ *
+ * @param expectedFile A snapshot file with the expected content
+ * @param actualFile The actual snapshot content
+ * @returns The diff output or `undefined`
+ */
+function tryRenderDiff(expectedFile: string, actualFile: string) {
+  const color = process.env.CI ? "never" : "always";
+  const out = cp.spawnSync("diff", ["-u", expectedFile, actualFile, `--color=${color}`]);
+  if (out.error) {
+    return undefined;
+  }
+
+  return out.stdout?.toString();
+}
+
+/**
+ * Creates a markdown snapshot of the synthesis output.
+ *
+ * @param baseName The base name of the test
+ * @param synthDir The directory containing the synthesis output
+ * @returns The snapshot content in markdown format
+ */
+async function createMarkdownSnapshot(baseName: string, synthDir: string) {
+  const fileMap: Record<string, string> = {};
+  const exclude = ["connections.json", "tree.json", "**/*.zip", "**/*.map"];
+  for await (const subpath of glob.iterate("**/*", {
```

Comment:
Is there a more specific glob we can make here? Maybe only js and json files?

Reply:
Yes we can start with this